R/assocTestAggregate.R

```diff
@@ -1,252 +1,264 @@
+              # pval.method <- match.arg(pval.method)
+
+              # don't use sparse matrices for imputed dosages
+              if (imputed) sparse <- FALSE
+
+              # coerce null.model if necessary
+              if (sparse) null.model <- .nullModelAsMatrix(null.model)
+              
+              # filter samples to match null model
+              sample.index <- .setFilterNullModel(gdsobj, null.model, verbose=verbose)
+
+              # do we need to match on alleles?
+              match.alleles <- any(c("ref", "alt") %in% names(mcols(currentRanges(gdsobj))))
+
+              # check ploidy
+              if (SeqVarTools:::.ploidy(gdsobj) == 1) male.diploid <- FALSE
+              
+              # results
+              res <- list()
+              res.var <- list()
+              i <- 1
+              n.iter <- length(variantFilter(gdsobj))
+              set.messages <- ceiling(n.iter / 100) # max messages = 100
+              iterate <- TRUE
+              while (iterate) {
+                  var.info <- variantInfo(gdsobj, alleles=match.alleles, expanded=TRUE)
+                  
+                  if (!imputed) {
+                      geno <- expandedAltDosage(gdsobj, use.names=FALSE, sparse=sparse)[sample.index,,drop=FALSE]
+                  } else {
+                      geno <- imputedDosage(gdsobj, use.names=FALSE)[sample.index,,drop=FALSE]
+                  }
+
+                  if (match.alleles) {
+                      index <- .matchAlleles(gdsobj, var.info)
+                      var.info <- var.info[index,,drop=FALSE]
+                      geno <- geno[,index,drop=FALSE]
+                  } else {
+                      index <- NULL
+                  }
+
+                  # number of non-missing samples
+                  n.obs <- colSums(!is.na(geno))
+                  
+                  # allele frequency
+                  freq <- .alleleFreq(gdsobj, geno, variant.index=index, sample.index=sample.index,
+                                      male.diploid=male.diploid, genome.build=genome.build)
+                  
+                  # filter monomorphic variants
+                  keep <- .filterMonomorphic(geno, count=n.obs, freq=freq$freq, imputed=imputed)
+
+                  # exclude variants with freq > max
+                  keep <-  keep & freq$freq <= AF.max
+                  if (!all(keep)) {
+                      var.info <- var.info[keep,,drop=FALSE]
+                      geno <- geno[,keep,drop=FALSE]
+                      n.obs <- n.obs[keep]
+                      freq <- freq[keep,,drop=FALSE]
+                  }
+
+                  # weights
+                  if (is.null(weight.user)) {
+                      # Beta weights
+                    if (burden.test %in% c("CMP", "BinomiRare")){
+                      weight <- seq(from=1, to=1, length.out=length(freq$freq))
+                    } else {
+                      weight <- .weightFromFreq(freq$freq, weight.beta)

```

Comment:
We probably need to make it explicit that CMP and BinomiRare ignore the `weight.beta` argument.